cilium/cmd/debuginfo.go

```diff
@@ -93,43 +180,85 @@ func runDebugInfo(cmd *cobra.Command, args []string) {
 		os.Exit(1)
 	}
 
-	// define output type and file path
 	var output outputType
-	var path string
-
-	switch {
-	case len(file) > 0: // Markdown file
-		output = MARKDOWN
-		path = file
-	case len(html) > 0: // HTML file
-		output = HTML
-		path = html
-	default: // Write to standard output
-		output = STDOUT
-	}
 
 	// create tab-writer to fill buffer
 	var buf bytes.Buffer
 	w := tabwriter.NewWriter(&buf, 5, 0, 3, ' ', 0)
 	p := resp.Payload
 
-	// generate multiple files
-	if (len(file) > 0 || len(html) > 0) && filePerCommand {
-		for cmdName, section := range sections {
+	cmdTime := time.Now()
+
+	if outputToFile && len(outputOpts) == 0 {
+		outputOpts = append(outputOpts, markdownOutput)
+	}
+
+	// Dump payload for each output format.
+	for _, outputOpt := range outputOpts {
+		var fileName string
+
+		switch strings.ToUpper(outputOpt) {
+		case markdownOutput:
+			output = MARKDOWN
+		case htmlOutput:
+			output = HTML
+		case jsonOutputDebuginfo:
+			output = JSONOUTPUT
+		case jsonpathOutput:
+			output = JSONPATH
+		default:
+			if re.MatchString(outputOpt) {
+				output = JSONPATH
+			}
+		}
+
+		if outputToFile {
+			fileName = formatFileName(outputDir, cmdTime, output)
+		}
+
+		// Generate multiple files for each subsection of the command if
+		// specified, except in the JSON cases.
```

Comment:
This comment describes exactly what the if statement code states (which is redundant), but doesn't describe the piece of information I'm curious about which is: Why is this not necessary in the JSON case?

Reply:
Will have a comment in the next upload. Basically I don't see a purpose of having separate files for each sub-field within the DebugInfo model.